internal/catchup_file_unwrapper.go

```diff
@@ -14,7 +14,11 @@ type CatchupFileUnwrapper struct {
 
 func (u *CatchupFileUnwrapper) UnwrapNewFile(reader io.Reader, header *tar.Header, file *os.File) error {
 	if u.options.isIncremented {
-		err := CreateFileFromIncrement(reader, file)
+		fileInfo, err := file.Stat()
```

Comment:
Instead of repeat this logic in each place you can inline it to NewReadWriterAtFrom